src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -568,7 +611,7 @@ async def compute_compile_action_id(
     compile_tool_id = await Get(GoSdkToolIDResult, GoSdkToolIDRequest("compile"))
     h.update(f"compile {compile_tool_id.tool_id}\n".encode())
     # TODO: Add compiler flags as per `go`'s algorithm. Need to figure out
```

Comment:
Old, but: TODO.

Reply:
Yes and won't be fixed until I obtain either your or Eric's advice on the rule graph errors in https://github.com/pantsbuild/pants/pull/16843 which adds a `GoBuildContext` type which will make adding such compiler flags much easier than currently feasible.